.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,16 @@
+name: deploy
+
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to Quay
+        run: docker login -u="${{ secrets.QUAY_USER }}" -p="${{ secrets.QUAY_SECRET }}" quay.io
+      - name: Push image to Quay
+        run: make IMAGE_TAG=main push
```

Comment:
nit: can we use the `docker/build-push-action` action to handle the build/push process for us?

Reply:
Only push-back I'll have on that is we'll need to define the image tag we're using in multiple places (I think) unless I'm missing something.

Reply:
We can't simply update the Makefile/plain+v0 provisioner manifests to use the main tag instead of the latest tag?

Reply:
I was meaning that the image tag that is defined in the Makefile would need to be defined in the action here, right? So at that point we have it written in the Makefile and this action which could lead to drift. Maybe that isn't an issue?

Reply:
Ah, I understand what you're saying now. I'm not sure I'm too worried about potential skew between what's defined in the Makefile and CI workflows.

Reply:
Sounds good! I updated the PR to use the official actions and also support pushing on `releases`.